frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.unit.spec.tsx

```diff
@@ -0,0 +1,139 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import type { Engine } from "metabase-types/api";
+
+import DatabaseForm from "./DatabaseForm";
+
+interface SetupResult {
+  onSubmit: () => void;
+}
+
+function setup(): SetupResult {
+  const onSubmit = jest.fn();
+  render(<DatabaseForm engines={ENGINES} isAdvanced onSubmit={onSubmit} />);
+
+  return { onSubmit };
+}
+
+describe("DatabaseForm", () => {
+  it("should submit default values", async () => {
+    const { onSubmit } = setup();
+
+    const expectedDatabaseName = "My H2 Database";
+    const expectedConnectionString = "file:/somewhere";
+    userEvent.type(screen.getByLabelText("Display name"), expectedDatabaseName);
+    userEvent.type(
+      screen.getByLabelText("Connection String"),
+      expectedConnectionString,
+    );
+
+    const saveButton = screen.getByRole("button", { name: "Save" });
+    await waitFor(() => {
+      expect(saveButton).toBeEnabled();
+    });
+    userEvent.click(saveButton);
+
+    await waitFor(() => {
+      expect(onSubmit).toHaveBeenCalledWith({
+        ...EXPECTED_DEFAULT_SCHEMA,
+        engine: "h2",
+        name: expectedDatabaseName,
+        details: {
+          "advanced-options": false,
+          db: expectedConnectionString,
+        },
+      });
+    });
+  });
+});
+
+const EXPECTED_DEFAULT_SCHEMA = {
```

Comment:
This asserts that the default value we have set is correct
https://github.com/metabase/metabase/blob/4534eeb1a717c81fe346a7de5d542aa27c59e7a2/frontend/src/metabase/databases/utils/schema.ts#L23-L38